dll/shellext/fontext/CFontExt.cpp

```diff
@@ -134,10 +138,83 @@ STDMETHODIMP CFontExt::GetDetailsOf(PCUITEMID_CHILD pidl, UINT iColumn, SHELLDET
         return E_FAIL;
     }
 
+    // Name, ReactOS specific?
+    if (iColumn == 0)
+        return GetDisplayNameOf(pidl, 0, &psd->str);
+
+    const FontPidlEntry* fontEntry = _FontFromIL(pidl);
+    if (!fontEntry)
+    {
+        ERR("ERROR, not a font PIDL!\n");
+        return E_FAIL;
+    }
+
+    // If we got here, we are in details view!
+    // Let's see if we got info about this file that we can re-use
+    if (m_LastDetailsFontName != fontEntry->Name)
+    {
+        CStringW File = g_FontCache->Filename(fontEntry, true);
+        HANDLE hFile = FindFirstFileW(File, &m_LastDetailsFileData);
+        if (hFile == INVALID_HANDLE_VALUE)
+        {
+            m_LastDetailsFontName.Empty();
+            ERR("Unable to query info about %S\n", File.GetString());
+            return HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+        }
+        FindClose(hFile);
+        m_LastDetailsFontName = fontEntry->Name;
+    }
+
+    // Most code borrowed from CFSFolder::GetDetailsOf
+    FILETIME lft;
+    SYSTEMTIME time;
+    int ret;
+    LARGE_INTEGER FileSize;
+    CStringA AttrLetters;
```

Comment:
Why does it use `CStringA` (ANSI)?

Reply:
Because the code is based on `_ILGetFileAttributes`, that is written like this to use the `STRRET`'s builtin `char      cStr[MAX_PATH];`